### PR TITLE
Terrargunt Migration: AS Staging

### PR DESCRIPTION
**Pull request by sinehome**
Pending

### CODE REVIEW

**Comment by sinehome** [comment]
Terraform will perform the following actions:

  # module.taskRouter.twilio_taskrouter_workspaces_workflows_v1.workflow["master"] will be updated in-place
  ~ resource "twilio_taskrouter_workspaces_workflows_v1" "workflow" {
      ~ configuration            = jsonencode( # whitespace changes
            {
                task_routing = {
                    filters = [
                        {
                            expression           = "channelType=='web' AND e2eTestMode=='true'"
                            filter_friendly_name = "E2E Test"
                            targets              = [
                                {
                                    expression = "(worker.waitingOfflineContact != true AND ((task.channelType == 'voice' AND worker.channel.chat.assigned_tasks == 0) OR (task.channelType != 'voice' AND worker.channel.voice.assigned_tasks == 0)) AND ((task.transferTargetType == 'worker' AND task.targetSid == worker.sid) OR (task.transferTargetType != 'worker' AND worker.sid != task.ignoreAgent))) OR (worker.waitingOfflineContact == true AND task.targetSid == worker.sid AND task.isContactlessTask == true)"
                                    queue      = "WQec0cfd11e72983e748585775e6826558"
                                },
                            ]
                        },
                        {
                            expression           = "channelType =='web'  OR isContactlessTask == true OR  twilioNumber == 'messenger:131329426738030'"
                            filter_friendly_name = "Aselo"
                            targets              = [
                                {
                                    expression = "(worker.waitingOfflineContact != true AND ((task.channelType == 'voice' AND worker.channel.chat.assigned_tasks == 0) OR (task.channelType != 'voice' AND worker.channel.voice.assigned_tasks == 0)) AND ((task.transferTargetType == 'worker' AND task.targetSid == worker.sid) OR (task.transferTargetType != 'worker' AND worker.sid != task.ignoreAgent))) OR (worker.waitingOfflineContact == true AND task.targetSid == worker.sid AND task.isContactlessTask == true)"
                                    queue      = "WQ55df346e2986acff362141c83f5cdf29"
                                },
                            ]
                        },
                    ]
                }
            }
        )
        id                       = "WSfe6fa22987a469a52948c19543f76d41/WWf0838456ca0de021769a04aee51e2d94"
        # (4 unchanged attributes hidden)
    }

  # module.taskRouter.twilio_taskrouter_workspaces_workflows_v1.workflow["survey"] will be updated in-place
  ~ resource "twilio_taskrouter_workspaces_workflows_v1" "workflow" {
      ~ configuration            = jsonencode( # whitespace changes
            {
                task_routing = {
                    filters = [
                        {
                            expression           = "isChatCaptureControl==true"
                            filter_friendly_name = "CaptureChannel"
                            targets              = [
                                {
                                    queue = "WQ92b6271e3922263d18d99136fc852894"
                                },
                            ]
                        },
                    ]
                }
            }
        )
        id                       = "WSfe6fa22987a469a52948c19543f76d41/WWd95bf5ba6269ccbc3ee936864a383b4a"
        # (4 unchanged attributes hidden)
    }

Plan: 0 to add, 2 to change, 0 to destroy.

**Comment by sinehome** [comment]
Configure Plan

Terraform will perform the following actions:

  # module.channel.twilio_studio_flows_v2.channel_studio_flow["facebook"] will be updated in-place
  ~ resource "twilio_studio_flows_v2" "channel_studio_flow" {
      ~ definition    = jsonencode( # whitespace changes
            {
                description   = "Facebook Studio Flow"
                flags         = {
                    allow_concurrent_calls = true
                }
                initial_state = "Trigger"
                states        = [
                    {
                        name        = "Trigger"
                        properties  = {
                            offset = {
                                x = 0
                                y = 0
                            }
                        }
                        transitions = [
                            {
                                event = "incomingMessage"
                                next  = "attributes"
                            },
                            {
                                event = "incomingCall"
                            },
                            {
                                event = "incomingConversationMessage"
                            },
                            {
                                event = "incomingRequest"
                            },
                            {
                                event = "incomingParent"
                            },
                        ]
                        type        = "trigger"
                    },
                    {
                        name        = "attributes"
                        properties  = {
                            attributes = <<-EOT
                                {
                                  "name": "{{trigger.message.ChannelAttributes.from}}",
                                  "channelType": "{{trigger.message.ChannelAttributes.channel_type}}",
                                  "channelSid": "{{trigger.message.ChannelSid}}",
                                  "twilioNumber": "{{trigger.message.ChannelAttributes.twilioNumber}}",
                                  "ignoreAgent": "",
                                  "language":"en-US",
                                  "transferTargetType": "",
                                  "memory": {{trigger.message.ChannelAttributes.memory| to_json}}
                                }
                            EOT
                            channel    = "TC43e2de5476f8234d53ae5d12b1f11a43"
                            offset     = {
                                x = 80
                                y = 250
                            }
                            workflow   = "WWf0838456ca0de021769a04aee51e2d94"
                        }
                        transitions = [
                            {
                                event = "callComplete"
                            },
                            {
                                event = "failedToEnqueue"
                            },
                            {
                                event = "callFailure"
                            },
                        ]
                        type        = "send-to-flex"
                    },
                ]
            }
        )
        id            = "FW9e965753250057ee7052c842c5aa0944"
        # (3 unchanged attributes hidden)
    }

  # module.channel.twilio_studio_flows_v2.channel_studio_flow["webchat"] will be updated in-place
  ~ resource "twilio_studio_flows_v2" "channel_studio_flow" {
      ~ definition    = jsonencode( # whitespace changes
            {
                description   = "Webchat Studio Flow"
                flags         = {
                    allow_concurrent_calls = true
                }
                initial_state = "Trigger"
                states        = [
                    {
                        name        = "Trigger"
                        properties  = {
                            offset = {
                                x = 0
                                y = 0
                            }
                        }
                        transitions = [
                            {
                                event = "incomingMessage"
                                next  = "attributes"
                            },
                            {
                                event = "incomingCall"
                            },
                            {
                                event = "incomingConversationMessage"
                            },
                            {
                                event = "incomingRequest"
                            },
                            {
                                event = "incomingParent"
                            },
                        ]
                        type        = "trigger"
                    },
                    {
                        name        = "attributes"
                        properties  = {
                            attributes = <<-EOT
                                {
                                  "e2eTestMode": "{{trigger.message.ChannelAttributes.pre_engagement_data.e2eTestMode}}",
                                  "ip": "{{trigger.message.ChannelAttributes.pre_engagement_data.ip}}",
                                  "firstName": "{{trigger.message.ChannelAttributes.from}}",
                                  "channelType": "{{trigger.message.ChannelAttributes.channel_type}}",
                                  "channelSid": "{{trigger.message.ChannelSid}}",
                                  "helpline": "{{trigger.message.ChannelAttributes.pre_engagement_data.helpline}}",
                                  "language":"en-US",
                                  "ignoreAgent": "",
                                  "transferTargetType": "",
                                  "preEngagementData": {{trigger.message.ChannelAttributes.pre_engagement_data | to_json}},
                                  "memory": {{trigger.message.ChannelAttributes.memory | to_json}}
                                }
                            EOT
                            channel    = "TC43e2de5476f8234d53ae5d12b1f11a43"
                            offset     = {
                                x = 80
                                y = 250
                            }
                            workflow   = "WWf0838456ca0de021769a04aee51e2d94"
                        }
                        transitions = [
                            {
                                event = "callComplete"
                            },
                            {
                                event = "failedToEnqueue"
                            },
                            {
                                event = "callFailure"
                            },
                        ]
                        type        = "send-to-flex"
                    },
                ]
            }
        )
        id            = "FWaaa7079ccdc2b97849895493a8e06a35"
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 2 to change, 0 to destroy.

File: twilio-iac/helplines/as/common.hcl [hunk 1]
```diff
@@ -3,51 +3,93 @@ locals {
...
-        bots       : local.en_us_bots
+    helpline                          = "Aselo"
+    old_dir_prefix                    = "aselo-as"
+    default_autopilot_chatbot_enabled = false
+    task_language                     = "en-US"
+    voice_ivr_language                = ""
+    enable_post_survey                = false
+
+    lex_bot_languages = {
+      en : ["pre_survey"]
```
Comment (janorivera) [inline twilio-iac/helplines/as/common.hcl:45]: I think you should use en_US  for this one

File: twilio-iac/helplines/as/common.hcl [hunk 2]
```diff
@@ -3,51 +3,93 @@ locals {
...
-  )
-
-  es_co_intents = merge(
-    jsondecode(file("/app/twilio-iac/helplines/configs/lex/es_CO/intents/pre_survey.json")),
-  )
-
-  es_co_bots = merge(
-    jsondecode(file("/app/twilio-iac/helplines/configs/lex/es_CO/bots/pre_survey.json")),
-  )
+  // en_us_slot_types = merge(
```
Comment (janorivera) [inline twilio-iac/helplines/as/common.hcl:6]: you can just remove these lines, no need to comment them out

File: webchat/configurations/as-staging.ts [hunk 3]
```diff
@@ -17,8 +17,8 @@
 import { Translations, Configuration, MapHelplineLanguage, ContactType } from '../types';
 import type { PreEngagementFormDefinition } from '../src/pre-engagement-form';
 
-const accountSid = 'AC6b99858a6faf7af1b572c83988b50eb1';
-const flexFlowSid = 'FO57c22d5dfc7a18dcada507aa70ca0cb3';
+const accountSid = 'AC1ca120bc71593bbe9ca78e2232a31e0d';
+const flexFlowSid = 'FWaaa7079ccdc2b97849895493a8e06a35';
```
Comment (janorivera) [inline webchat/configurations/as-staging.ts:21]: This should not be the studio flow, it should be the flex flow sid, they usually start with "FO...."

**Review by janorivera — CHANGES_REQUESTED** [review]

**Pull request merged**